react-common/components/palette/ColorPickerField.tsx

```diff
@@ -42,24 +21,30 @@ export const ColorPickerField = (props: ColorPickerFieldProps) => {
     }
 
     const onColorPickerChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setCurrentColor(e.target.value);
+        setCurrentColor(e.target.value.toUpperCase());
     }
 
     const onTextInputChanged = (newValue: string) => {
+        if (newValue?.[0] != '#') {
+            newValue = "#" + newValue;
+        }
+        if (newValue.length > 7) {
+            newValue = newValue.substring(0,7);
+        }
         if (/#[0-9a-fA-F]{6}/.test(newValue)) {
-            onColorChanged(newValue);
+            onColorChanged(newValue.toUpperCase());
         }
```

Comment:
I experience some weirdness with this functionality on the first color in the list. If I try to enter any text, it doesn't acknowledge it. This works fine with the other colors in the list. 

https://user-images.githubusercontent.com/49178322/206545046-60389988-0a72-463b-9389-e2ad4ab05c97.mp4

Reply:
I'm also a bit worried about an edge case where users just spam the keyboard to see what happens in the text field and the color that was there is lost. 

However, I also think that if users are wanting to experiment like that, then it's fair for that progress to be lost. This might be something that we just want to see if users get upset about and change the logic later.

Just wanted to comment as something we might want to keep in mind.

Reply:
I'm having trouble reproducing the bug with the first color. @jwunderl was this happening for you as well?

> I'm also a bit worried about an edge case where users just spam the keyboard to see what happens in the text field and the color that was there is lost.

Yea, ideally the text input wouldn't let you write any non-hex characters. I'm not sure why currently the text validation doesn't happen until on blur. But if the value is invalid, it should change back to the previous value.